### PR TITLE
[REF] point_of_sale, pos_*: only use export_for_ui for serializing

### PR DESCRIPTION
**Pull request by davidmonnom**
*:pos_hr,pos_hr_restaurant,pos_loyalty,pos_restaurant

Before when serializing a pos.order from the server, we are using
several methods for different use cases.

- table syncing: get_table_draft_orders
- order sharing: get_draft_share_order_ids
- ticket screen: export_for_ui

All of these methods had their own logic.

Now, to improve the maintainability of point_of_sale, order
serialization is only performed via export_for_ui.

Enterprise PR: https://github.com/odoo/enterprise/pull/45151
TaskId: 3449870

### CODE REVIEW

File: addons/pos_loyalty/models/pos_order.py [hunk 1]
```diff
@@ -145,21 +145,6 @@ def confirm_coupon_programs(self, coupon_data):
...
-        if order_line.get("reward_id"):
-            order_line["reward_id"] = order_line["reward_id"][0]
-        if order_line.get("coupon_id"):
-            order_line["coupon_id"] = order_line["coupon_id"][0]
-
-        return order_line
-
-    def _get_fields_for_order_line(self):
-        fields = super(PosOrder, self)._get_fields_for_order_line()
-        fields.extend(['is_reward_line', 'reward_id', 'coupon_id', 'reward_identifier_code', 'points_cost'])
```
Comment (caburj) [inline addons/pos_loyalty/models/pos_order.py:160]: include these fields in the _export_for_ui of pos.order.line

**Review by caburj — COMMENTED** [review]
Ignore the comment if they are already listed in the _export_for_ui.

File: addons/pos_hr/models/pos_order.py [hunk 2]
```diff
@@ -28,17 +28,3 @@ def _export_for_ui(self, order):
             'employee_id': order.employee_id.id,
         })
         return result
-
-    def _get_fields_for_draft_order(self):
-        fields = super(PosOrder, self)._get_fields_for_draft_order()
-        fields.append('employee_id')
```
[the remark this answers is not in the document]
Comment (davidmonnom) [reply addons/pos_hr/models/pos_order.py:34]: This is the case just above.

File: addons/pos_restaurant/models/pos_order.py [hunk 3]
```diff
@@ -21,39 +21,6 @@ class PosOrder(models.Model):
...
     customer_count = fields.Integer(string='Guests', help='The amount of customers that have been served by this order.', readonly=True)
 
-    def _get_fields_for_order_line(self):
-        fields = super(PosOrder, self)._get_fields_for_order_line()
-        fields.extend([
-            'note',
-            'full_product_name',
-            'customer_note',
-            'price_extra',
-            'refunded_orderline_id',
```
[the remark this answers is not in the document]
Comment (davidmonnom) [reply addons/pos_restaurant/models/pos_order.py:31]: Already in PoS

File: addons/point_of_sale/models/pos_order.py [hunk 4]
```diff
@@ -1111,6 +1111,7 @@ def _export_for_ui(self, order):
             'state': order.state,
             'account_move': order.account_move.id,
             'id': order.id,
+            'server_id': order.id,
```
Comment (davidmonnom) [inline addons/point_of_sale/models/pos_order.py:1114]: Was handled in the frontend 
``this.server_id = json.server_id || json.id || false;``

File: addons/point_of_sale/models/pos_order.py [hunk 5]
```diff
@@ -1298,6 +1299,7 @@ def _export_for_ui(self, orderline):
             'discount': orderline.discount,
             'tax_ids': [[6, False, orderline.tax_ids.mapped(lambda tax: tax.id)]],
             'id': orderline.id,
+            'server_id': orderline.id,
```
Comment (davidmonnom) [inline addons/point_of_sale/models/pos_order.py:1302]: Really used for orderlines ?

File: addons/point_of_sale/models/pos_order.py (hunk 4 above)
Comment (caburj) [reply addons/point_of_sale/models/pos_order.py:1114]: oh, it means that during initialization of the order in the frontend, the "id" is used as server_id, and we don't need to put "server_id" in the export for ui. Maybe we should do same logic for other pos frontend models.

Comment (caburj) [reply addons/point_of_sale/models/pos_order.py:1114]: I'll remove and update the init_from_JSON of the pos ui models.

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**